credentials/credentials.go

```diff
@@ -215,6 +215,21 @@ func RequestInfoFromContext(ctx context.Context) (ri RequestInfo, ok bool) {
 	return ri, ok
 }
 
+// NewRequestInfoContext creates a new context with the given RequestInfo
+// attached to it.
+//
+// This RequestInfo will be accessible via RequestInfoFromContext.
+//
+// Intended to be used from tests for PerRPCCredentials implementations (that
+// often need to check connection's SecurityLevel). Should not be used from
+// non-test code: the gRPC client already prepares a context with the correct
+// RequestInfo attached when calling PerRPCCredentials.GetRequestMetadata.
+//
+// This API is experimental.
+func NewRequestInfoContext(ctx context.Context, ri RequestInfo) context.Context {
```

Comment:
`NewContextWithRequestInfo` seems like it would be more natural.  Also matches e.g. https://pkg.go.dev/google.golang.org/grpc#NewContextWithServerTransportStream

Reply:
Done.